sdk/python/kfp/compiler/compiler.py

```diff
@@ -832,8 +832,8 @@ def _create_pipeline_workflow(self,
 
         # set ttl after workflow finishes
         if pipeline_conf.ttl_seconds_after_finished >= 0:
-            workflow['spec'][
-                'ttlSecondsAfterFinished'] = pipeline_conf.ttl_seconds_after_finished
+            workflow['spec']['ttlStrategy'] = {}
```

Comment:
nit: the following line can be combined into this one as 
`workflow['spec']['ttlStrategy'] = {'secondsAfterCompletion' = pipeline_conf.ttl_seconds_after_finished}`